File: C++/Factorial.cpp
#include<iostream>
using namespace std;

int fact(int n){
    if(n==0){
        return 1;
    }
    return n*fact(n-1);
}

int main(){
    int n,res;
    cout<<"Enter the number : ";
    cin>>n;
    res=fact(n);
    cout<<"Factorial of "<<n<<" is : "<<res;
    return 0;
}

File: Java/Factorial.java
package com.lev;

import java.util.Scanner;

public class Factorial {
    static int factorial(int n){
        if(n == 0){
            return 1;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the number : ");
        int n = input.nextInt();
        System.out.println("Factorial of " + n + " is " + factorial(n));
    }
}
